examples/simulation/simulate_telescope.cpp

```diff
@@ -79,11 +81,17 @@ int simulate(std::string output_directory, unsigned int events,
      ***************************/
 
     // Origin of particles
-    auto generator =
+    using generator_type =
         detray::random_track_generator<traccc::free_track_parameters,
-                                       uniform_gen_t>(
-            pg_opts.gen_nparticles, pg_opts.vertex, pg_opts.vertex_stddev,
-            pg_opts.mom_range, pg_opts.theta_range, pg_opts.phi_range);
+                                       uniform_gen_t>;
+    generator_type::configuration gen_cfg{};
```

Comment:
Have you considered putting the `generator_type::configuration` as a member into the `particle_gen_options` ?

Reply:
Let's just go like this